crates/common/log_manager/src/log_utils.rs

```diff
@@ -93,42 +93,73 @@ pub fn read_log_content(
     }
 }
 
-pub fn filter_logs_on_type(
-    files: &Vec<FileEntry>,
+fn filter_logs(
+    files: &[FileEntry],
     log_type: &str,
+    date_from: OffsetDateTime,
 ) -> Result<Vec<PathBuf>, LogRetrievalError> {
-    let mut files_to_send = Vec::new();
-    for file in files {
-        let maybe_file_path = file.path.as_str(); // because it can be a glob pattern
-        let file_type = file.config_type.as_str();
-
-        if !file_type.eq(log_type) {
-            continue;
-        } else {
-            for entry in glob(maybe_file_path)? {
-                let file_path = entry?;
-                files_to_send.push(file_path)
-            }
-        }
-    }
-    if files_to_send.is_empty() {
+    let file_list = filter_logs_by_type(files, log_type)?;
+
+    let file_list = filter_logs_by_date(file_list, date_from);
+
+    if file_list.is_empty() {
         Err(LogRetrievalError::NoLogsAvailableForType {
             log_type: log_type.to_string(),
         })
     } else {
-        Ok(files_to_send)
+        Ok(sort_logs_by_date(file_list))
     }
 }
 
-/// filter a vector of pathbufs according to `obj.log.date_from` and `obj.log.date_to`
-pub fn filter_logs_path_on_metadata(
+fn filter_logs_by_type(
+    files: &[FileEntry],
     log_type: &str,
+) -> Result<Vec<(PathBuf, OffsetDateTime, bool)>, LogRetrievalError> {
+    let mut file_list = Vec::new();
+    let wildcard_regex = Regex::new(r"^.*\*.*").unwrap();
+
+    let files: Vec<&str> = files
+        .iter()
+        .filter(|file| file.config_type.eq(log_type))
+        .map(|file| file.path.as_str())
+        .collect();
+
+    for file in files {
+        let paths = glob(file)?;
+        for path in paths {
+            let entry = path?;
+            file_list.push((
+                entry.to_owned(),
+                get_modification_date(entry.as_path())?,
+                wildcard_regex.is_match(file),
+            ))
+        }
+    }
+
+    Ok(file_list)
+}
+
+fn get_modification_date(path: impl AsRef<Path>) -> std::io::Result<OffsetDateTime> {
+    let metadata = std::fs::metadata(&path)?;
+    Ok(OffsetDateTime::from(metadata.modified()?))
+}
+
+fn filter_logs_by_date(
+    files: Vec<(PathBuf, OffsetDateTime, bool)>,
     date_from: OffsetDateTime,
-    mut logs_path_vec: Vec<PathBuf>,
-) -> Result<Vec<PathBuf>, LogRetrievalError> {
-    let mut out = vec![];
+) -> Vec<PathBuf> {
+    // include log file with static path no matter whether or not it is in date range
+    files
+        .into_iter()
+        .filter(|(_, modification_time, wildcard_match)| {
+            !wildcard_match || *modification_time >= date_from
+        })
+        .map(|(path, _, _)| path)
```

Comment:
I would do the sort by modification date here, before discarding the modification date that has already been extracted.